### PR TITLE
Adding a PR leg to run our style check and bringing native code into style compliance

### PR DESCRIPTION
**Pull request by sokket**
This PR has two different commits:

1) Moving the format-code.sh script over to python so we can use the same script on every platform without special shell-specific scripts. This python script works in two different ways; first, it can be used like format-code.sh right now and updates files to be in style compliance. The second way is by adding the ```checkonly``` parameter, in which case it will only check if all files are in compliance and return -1 when it finds one that is not.

2) Running the script on the native code to fix style issues

PTAL @stephentoub @mmitche

### CODE REVIEW

**Comment by ellismg** [comment]
FWIW, I don't really like the adoption of python here.  We already use shell for *NIX and CMD/PowerShell for Windows in the rest of our stuff, I don't really like adding yet another dependency (even if python is installed out of the box on most NIX's).

What's the value here?  So I can code-format on Windows?  Seems like it would be better just to write the small powershell script if we cared about that.

I'm a bit of a curmudgeon about this sort of stuff, so take that for what you will.

**Comment by sokket** [comment]
@ellismg I went with python here since it means we have one single script that runs on every platform in order to format. To get styling working on Windows, I would need to write yet another script that is OS specific; then, if we ever changed how we format or the parameters (etc), we would have to remember to change two files...leading to the potential of two different formatting tools. I'm a big fan of single, unified scripts like this more instead of a combination of multiple single-platform scripts (bash, PS, etc) especially since Helix has a dependency on Python anyway, but that's a personal preference.

**Comment by Clockwork-Muse** [comment]
@sokket - uh, isn't Helix a Microsoft-internal tool?  I don't have python installed on my machine (as a separate runtime), which would still leave me out in the cold, right?  Or is it somehow compiled and run on the CLR?  Granted, I haven't been messing with native code at all...

(Personally, I too would like to avoid platform-specific scripts, but I'd probably have chosen C#...)

**Comment by mmitche** [comment]
test ci please

File: netci.groovy [hunk 1]
```diff
@@ -27,6 +27,18 @@ def prCCJob = job(Utilities.getFullJobName(project, 'code_coverage_windows', tru
   }
 }
 
+[true, false].each { isPR -> 
+  def codeFormatterJobName = Utilities.getFullJobName(project, 'native_code_format_check', isPR)
+  def codeFormatterJob = job(codeFormatterJobName) {
+    label('ubuntu')
+    seteps {
```
Comment (mmitche) [inline netci.groovy]: seteps -> steps.  Comment 'test ci please' when fixed to rerun the CI definition generation test

File: src/Native/System.Native/pal_interfaceaddresses.cpp [hunk 2]
```diff
@@ -44,8 +44,7 @@ extern "C" int32_t EnumerateInterfaceAddresses(IPv4AddressFound onIpv4Found,
         {
             // IP Address
             IpAddressInfo iai = {
-                .InterfaceIndex = interfaceIndex,
-                .NumAddressBytes = NUM_BYTES_IN_IPV4_ADDRESS,
+                .InterfaceIndex = interfaceIndex, .NumAddressBytes = NUM_BYTES_IN_IPV4_ADDRESS,
```
Comment (mellinoe) [inline src/Native/System.Native/pal_interfaceaddresses.cpp]: Do we like how this is formatted now? I kind of prefer it on two lines, it's a little bit easier to read I think. Might just be a preference, though, and I don't have as much experience looking at C++ code :smile:

Comment (sokket) [inline src/Native/System.Native/pal_interfaceaddresses.cpp]: Last time this was brought up, I think everyone agreed that it looks better on multiple lines; however, I couldn't find the right option to do that. Also, any changes to the formatting style guidelines would be in a different PR :)

File: netci.groovy (hunk 1 above)
Comment (sokket) [inline netci.groovy]: Fixed

**Comment by sokket** [comment]
test ci please

**Comment by sokket** [comment]
test ci please

**Comment by sokket** [comment]
@Clockwork-Muse Python would only be required if you wanted to run the script; you can still format the native file(s) that you edit by running ```clang-format -style=file -i <file>``` yourself :). Since Python comes pre-installed on OS X and Ubuntu (and we install it on our Windows build machines) it seemed like the best x-plat choice

**Comment by mmitche** [comment]
LGTM on CI definition side.

**Comment by sokket** [comment]
@stephentoub Any feedback?

**Comment by stephentoub** [comment]
Other than Matt's feedback about choice of language, LGTM.  I don't have strong feelings one way or the other on that aspect, so I'll defer to Matt.  Thanks.

**Comment by stephentoub** [comment]
That said, I would have expected to see the leg mentioned here in the Jenkins job, but I don't see it:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_prtest/5972/console
Does it not show up until after this is merged, or am I just missing it?

**Comment by mmitche** [comment]
Won't up until merged. "test ci please" only ensures that the job can be generated.  In this case, it looked great to me.

**Comment by ellismg** [comment]
Jon and I talked about the Python stuff a little bit face to face.  Here are my two major objections:

1. I would like to prevent us from unnecessarily depending on software which is not commonly installed  one the platforms we are running on.
2. I would like us to when possible minimize the number of different scripting languages and tools we depend on.

To point (1) it means I would not like us to adopt python on windows.  I think PowerShell is a better fit for most scripting we'll need to do.  Since we can't do code formatting on Windows today (and we don't need to, since we don't write native code on Windows), I'm fine with python from that angle.

As for (2) if we take this change, it means that we should seriously consider taking our existing bash scripts and converting them to python so I don't have to know both bash and python to effectively work in this repository.  Jon said he would open an issue about this.  I think that will be a good place to decide if we want to "bet the farm" on Python as our *NIX scripting solution or do something else.

For now, I'm okay merging this change.  It isn't the way I would do it, but Jon feels strongly that this is the right direction to be moving and I'm fine with us making progress as long as we figure out a long term plan for all the scripts and put them into action.  The rest of the change (actually getting the legs) is real goodness. Thanks a great deal for doing this @sokket and putting up with curmudgeons like me that hate everything :-).

**Comment by sokket** [comment]
@ellismg Thanks Matt! I opened issue #4213 to chat about this and nail down a plan. If we decide to not go that route then I'll be more than happy to reverse this and make the script a bash (or whatever we decide to use) script.
